support/cas-server-support-saml-idp-metadata-redis/src/main/java/org/apereo/cas/config/SamlIdPRedisIdPMetadataConfiguration.java

```diff
@@ -36,7 +36,7 @@
  */
 @Configuration("SamlIdPRedisIdPMetadataConfiguration")
 @EnableConfigurationProperties(CasConfigurationProperties.class)
-@ConditionalOnProperty(prefix = "cas.authn.saml-idp.metadata.redis", name = "idp-metadata-enabled", havingValue = "true")
+@ConditionalOnProperty(prefix = "cas.authn.saml-idp.metadata.redis", name = {"idp-metadata-enabled", "enabled"}, matchIfMissing = true)
 @Slf4j
```

Comment:
This might not be 100% backwards compatible? Does matchIfMissing = true mean that if `cas.authn.saml-idp.metadata.redis.idp-metadata-enabled` is not set that it will be considered true even though it defaults to false?

Reply:
I think it's best to create two separate conditions and allow each to have its own matchIfMissing rule. As you note, it's rather confusing to reason about this rule with certainty.

Reply:
That annotation only allows a single instance to be used. I could try using `@ConditionalOnExpression`, although I think the way it is set now would require both properties to be true and would not create the configuration class if either were false.